msg/versioned/ArmingCheckReply.msg

```diff
@@ -1,34 +1,42 @@
+# Arming check reply.
+# 
+# This is a response to an ArmingCheckRequest message sent by the FMU to an external component, such as a ROS 2 navigation mode.
+# The response contains the current set of external mode requirements, and a queue of events indicating recent failures to set the mode (which the FMU may then forward to a ground station).
+# The request is sent regularly to all registered ROS modes, even while armed, so that the FMU always knows and can forward the current state.
+#
+# Note that the external component is identified by its registration_id, which is allocated to the component during registration (arming_check_id in RegisterExtComponentReply).
+# The message is not used by internal/FMU components, as their mode requirements are known at compile time.
+
 uint32 MESSAGE_VERSION = 0
 
-uint64 timestamp # time since system start (microseconds)
+uint64 timestamp # [us] Time since system start.
 
-uint8 request_id
-uint8 registration_id
+uint8 request_id # Id of ArmingCheckRequest for which this is a response.
+uint8 registration_id # Id of external component emitting this response.
 
-uint8 HEALTH_COMPONENT_INDEX_NONE = 0
+uint8 HEALTH_COMPONENT_INDEX_NONE = 0 # Index of health component for which this response applies.
 
-uint8 health_component_index      # HEALTH_COMPONENT_INDEX_*
-bool health_component_is_present
-bool health_component_warning
-bool health_component_error
+uint8 health_component_index # [@enum HEALTH_COMPONENT_INDEX]
+bool health_component_is_present # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_warning # Unused. Intended for use with health events interface (health_component_t in events.json).
+bool health_component_error # Unused. Intended for use with health events interface (health_component_t in events.json).
 
-bool can_arm_and_run              # whether arming is possible, and if it's a navigation mode, if it can run
+bool can_arm_and_run # True if the component can arm. For navigation mode components, true if the component can arm in the mode or switch to the mode when already armed.
```

Comment:
@bkueng  I am still not convinced by this. Your original was "if it can run", which implies that this is the clause that means "or switch to the mode if it is already armed"`.

It doesn't have to mean that. You could have been referring to the case where something is armed but still interlocked. I'm thinking here of a throw launch where the vehicle is armed in the mode or can switch to the mode but in a particular state where it is not active until an acceleration trigger is detected.
But I will assume not!

Reply:
Yes it's for switching into the mode.